diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/DataClassesRule.kt

```diff
@@ -56,15 +56,26 @@ class DataClassesRule(configRules: List<RulesConfig>) : DiktatRule("data-classes
     @Suppress("UnsafeCallOnNullableType", "FUNCTION_BOOLEAN_PREFIX", "ComplexMethod")
     private fun ASTNode.canBeDataClass(): Boolean {
         val isNotPropertyInClassBody = findChildByType(CLASS_BODY)?.let { (it.psi as KtClassBody).properties.isEmpty() } ?: true
+        val constructorParametersNames = mutableListOf<String>()
         val hasPropertyInConstructor = findChildByType(PRIMARY_CONSTRUCTOR)
             ?.let { constructor ->
                 (constructor.psi as KtPrimaryConstructor)
                     .valueParameters
-                    .run { isNotEmpty() && all { it.hasValOrVar() } }
+                    .run { isNotEmpty() && all { if (it.hasValOrVar()) true else { constructorParametersNames.add(it.name!!); false } } }
```

Comment:
I'd move this `add` into separate `onEach` before `run`, now it's not very readable

Reply:
Fixed